builder/builder.go

```diff
@@ -522,137 +529,156 @@ func (b *Builder) BuildChroots(template *x509.Certificate, privkey *rsa.PrivateK
 	return nil
 }
 
-// Set the published versions to what was just built
-func (b *Builder) setVersion(publish bool) {
-	if publish == false {
-		return
-	}
+// BuildUpdate will produce an update consumable by the swupd client
+func (b *Builder) BuildUpdate(prefixflag string, minVersion int, format string, skipSigning bool, publish bool, keepChroots bool) error {
+	var timer stopWatch
+	defer timer.WriteSummary(os.Stdout)
 
-	// Create the www/version/format# dir if it doesn't exist
-	formatdir := b.Statedir + "/www/version/format" + b.Format
-	if _, err := os.Stat(formatdir); os.IsNotExist(err) {
-		if err = os.MkdirAll(formatdir, 0777); err != nil {
-			helpers.PrintError(err)
-			os.Exit(1)
-		}
+	if format != "" {
+		b.Format = format
 	}
 
-	fmt.Println("Setting latest version to " + b.Mixver)
-	err := ioutil.WriteFile(formatdir+"/latest", []byte(b.Mixver), 0644)
+	// Ensure the format dir exists.
+	formatDir := filepath.Join(b.Statedir, "www", "version", "format"+b.Format)
+	err := os.MkdirAll(formatDir, 0777)
 	if err != nil {
-		helpers.PrintError(err)
-		os.Exit(1)
+		return errors.Wrapf(err, "couldn't create the format directory")
 	}
 
-	err = ioutil.WriteFile(b.Statedir+"/image/LAST_VER", []byte(b.Mixver), 0644)
+	// Create update metadata for the mix.
+	timer.Start("create update")
+	updatecmd := exec.Command(prefixflag+"swupd_create_update", "-S", b.Statedir, "--minversion", strconv.Itoa(minVersion), "-F", b.Format, "--osversion", b.Mixver)
```

Comment:
Why not also use the option to run the new `CreateManifests()` function as well?

Reply:
I'm working on those :-)